staging/src/k8s.io/apimachinery/pkg/util/httpstream/wsstream/conn_test.go

```diff
@@ -271,3 +273,136 @@ func TestVersionedConn(t *testing.T) {
 		}()
 	}
 }
+
+func TestIsWebSocketRequestWithStreamCloseProtocol(t *testing.T) {
+	tests := map[string]struct {
+		headers  map[string][]string
+		expected bool
+	}{
+		"No headers returns false": {
+			headers:  map[string][]string{},
+			expected: false,
+		},
+		"Only connection upgrade header is false": {
+			headers: map[string][]string{
+				"Connection": {"upgrade"},
+			},
+			expected: false,
+		},
+		"Only websocket upgrade header is false": {
+			headers: map[string][]string{
+				"Upgrade": {"websocket"},
+			},
+			expected: false,
+		},
+		"Only websocket and connection upgrade headers is false": {
+			headers: map[string][]string{
+				"Connection": {"upgrade"},
+				"Upgrade":    {"websocket"},
+			},
+			expected: false,
+		},
+		"Missing connection/upgrade header is false": {
+			headers: map[string][]string{
+				"Upgrade":               {"websocket"},
+				WebSocketProtocolHeader: {"v5.channel.k8s.io"},
+			},
+			expected: false,
+		},
+		"Websocket connection upgrade headers with v5 protocol is true": {
+			headers: map[string][]string{
+				"Connection":            {"upgrade"},
+				"Upgrade":               {"websocket"},
+				WebSocketProtocolHeader: {"v5.channel.k8s.io"},
+			},
+			expected: true,
+		},
+		"Websocket connection upgrade headers with wrong case v5 protocol is false": {
+			headers: map[string][]string{
+				"Connection":            {"upgrade"},
+				"Upgrade":               {"websocket"},
+				WebSocketProtocolHeader: {"v5.CHANNEL.k8s.io"}, // header value is case-sensitive
+			},
+			expected: false,
+		},
+		"Websocket connection upgrade headers with v4 protocol is false": {
+			headers: map[string][]string{
+				"Connection":            {"upgrade"},
+				"Upgrade":               {"websocket"},
+				WebSocketProtocolHeader: {"v4.channel.k8s.io"},
+			},
+			expected: false,
+		},
+		"Websocket connection upgrade headers with multiple protocols but missing v5 is false": {
+			headers: map[string][]string{
+				"Connection":            {"upgrade"},
+				"Upgrade":               {"websocket"},
+				WebSocketProtocolHeader: {"v4.channel.k8s.io", "v3.channel.k8s.io", "v2.channel.k8s.io"},
+			},
+			expected: false,
+		},
+		"Websocket connection upgrade headers with multiple protocols including v5 is true": {
+			headers: map[string][]string{
+				"Connection":            {"upgrade"},
+				"Upgrade":               {"websocket"},
+				WebSocketProtocolHeader: {"v5.channel.k8s.io", "v4.channel.k8s.io"},
+			},
+			expected: true,
+		},
+		"Websocket connection upgrade headers with multiple protocols out of order including v5 is true": {
+			headers: map[string][]string{
+				"Connection":            {"upgrade"},
+				"Upgrade":               {"websocket"},
+				WebSocketProtocolHeader: {"v4.channel.k8s.io", "v5.channel.k8s.io", "v3.channel.k8s.io"},
```

Comment:
add a test case with "sec-websocket-protocol" if my understanding of the protocol is correct  https://github.com/kubernetes/kubernetes/pull/119186/files#r1320508587

Reply:
Added test.